src/lib/y2storage/dialogs/guided_setup/select_root_disk.rb

```diff
@@ -93,31 +119,71 @@ def linux_actions_widget
         end
 
         def any_disk_widget
-          Left(RadioButton(Id(:any), _("Any disk")))
+          Left(RadioButton(Id(:any_disk), Opt(:notify), _("Any disk")))
         end
 
         def disk_widget(disk)
-          Left(RadioButton(Id(disk.name), disk_label(disk)))
+          Left(RadioButton(Id(disk_id(disk)), Opt(:notify), disk_label(disk)))
         end
 
         def initialize_widgets
-          widget = settings.root_device || :any
-          widget_update(widget, true)
+          if need_to_select_disk?
+            root = settings.root_device
+            widget = root ? disk_id(root) : :any_disk
+            widget_update(widget, true)
+          end
+          root_disk_handler
         end
 
         def update_settings!
-          root = candidate_disks.detect { |d| widget_value(d.name) }
+          root = selected_disk
           settings.root_device = root ? root.name : nil
-          true
         end
 
       private
 
+        def define_disks_handlers
+          options = candidate_disks.map { |d| disk_id(d) }
+          options = options.unshift(:any_disk)
+          options.each do |option|
+            define_singleton_method :"#{option}_handler" do
+              root_disk_handler
+            end
+          end
+        end
+
+        def disk_id(disk)
+          name = disk.is_a?(String) ? disk : disk.name
+          name.split("/").last
```

Comment:
Maybe I'm being a little bit paranoid but I think we should add an explicit documentation somewhere stating that we use just the last part of the path as widget id. Sure there is no collision risk using regular disk names like `/dev/sda`, but... you know :wink: